backend/de.metas.printing/de.metas.printing.base/src/main/java/de/metas/printing/api/impl/AbstractPrintingDAO.java

```diff
@@ -257,8 +276,12 @@ public final I_AD_Printer_Matching retrievePrinterMatchingOrNull(
 			@Nullable final UserId userToPrintId,
 			@NonNull final de.metas.adempiere.model.I_AD_Printer printer)
 	{
+		final WorkplaceService workplaceService = SpringContextHolder.instance.getBean(WorkplaceService.class);
```

Comment:
please, move this definition at the top of, with the other services.

Reply:
@riadmen  @metas-rc (leave it like this now)
but calling a service (final WorkplaceService workplaceService) from a Repository/DAO is not a good practice at all!
That's a sign that workplaceId should be provided as parameter